### PR TITLE
config cleanup & implement free_config

### PR DESCRIPTION
**Pull request by martinetd**
Some typos but it's mostly leak fixing, was waiting for bindings to land to push

### CODE REVIEW

**Comment by 4e554c4c** [comment]
I know that some of these lists contain pointers that should be freed too. Should we just add that when implementing those parts of the config?

**Comment by martinetd** [comment]
Yes, there is no implementation currently so the various free functions are in headers but do not exist yet.
I have an example with cmd_set I'll push after this; a problem of this approach is that it serializes things quite a bit... We can add other cleanups later if that's a problem.

File: sway/commands/input.c [hunk 1]
```diff
@@ -24,7 +24,11 @@ struct cmd_results *cmd_input(int argc, char **argv) {
 	char **argv_new = argv+2;
 
 	struct cmd_results *res;
+	struct input_config *old_input_config = current_input_config;
 	current_input_config = new_input_config(argv[0]);
+	if (!current_input_config) {
+		return cmd_results_new(CMD_FAILURE, NULL, "Couldn't allocate config");
```
Comment (emersion) [inline sway/commands/input.c]: If we can't allocate an input config, we probably shouldn't allocate a command result.

Comment (martinetd) [reply sway/commands/input.c]: Yeah, I thought about that, but I have no idea how to return something else? We can't return something on the stack or a null..

Comment (SirCmpwn) [reply sway/commands/input.c]: This is an unsolved problem bigger than this PR.

**Comment by SirCmpwn** [comment]
We should implement the reload command soon.

**Comment by martinetd** [comment]
I've got set in the pipe, can do reload as well this weekend.

**Pull request merged**